--- local_lib.php-	2021-10-29 14:34:46.986915441 +0900
+++ local_lib.php	2021-10-29 19:15:20.912390494 +0900
@@ -908,6 +908,13 @@
         $requestparams['lis_person_name_family'] = $USER->lastname;
         $requestparams['lis_person_name_full'] = fullname($USER);
         $requestparams['ext_user_username'] = $USER->username;
+        //
+        $userinfo = posix_getpwnam($USER->username);
+        if ($userinfo) {
+            $requestparams['ext_user_userid']    = $userinfo['uid'];
+            $requestparams['ext_user_groupid']   = $userinfo['gid'];
+            $requestparams['ext_user_groupname'] = posix_getgrgid($userinfo['gid'])['name'];
+        }
     }
 
     if ($islti2 || $typeconfig['sendemailaddr'] == LTI_SETTING_ALWAYS ||
